src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Tests/Controllers/DataSetsControllerTests.cs

```diff
@@ -703,4 +707,479 @@ private async Task<HttpResponseMessage> GetVersion(Guid dataSetId, string dataSe
+
+            var locationMetas = new List<LocationMeta>();
+
+            foreach (var locationOptionMetaGenerator in allLocationOptionMetaTypesGeneratorByLevel)
+            {
+                LocationMeta locationMeta = DataFixture
+                    .DefaultLocationMeta()
+                    .WithLevel(locationOptionMetaGenerator.Key);
+
+                var locationOptionMetas = new List<LocationOptionMeta>()
+                {
+                    locationOptionMetaGenerator.Value.Invoke(),
+                    locationOptionMetaGenerator.Value.Invoke(),
+                    locationOptionMetaGenerator.Value.Invoke(),
+                };
+
+                foreach (var locationOptionMeta in locationOptionMetas)
+                {
+                    var locationOptionMetaLink = DataFixture
+                        .DefaultLocationOptionMetaLink()
+                        .WithMeta(locationMeta)
+                        .Generate();
+
+                    locationOptionMeta.MetaLinks.Add(locationOptionMetaLink);
+                }
+
+                locationMeta.Options.AddRange(locationOptionMetas);
+
+                locationMetas.Add(locationMeta);
+            }
+
+            DataSetVersion dataSetVersion = DataFixture
+                .DefaultDataSetVersion()
+                .WithStatusPublished()
+                .WithDataSetId(dataSet.Id)
+                .WithFilterMetas(() => filterMetas)
+                .WithLocationMetas(() => locationMetas)
+                .WithIndicatorMetas(() =>
+                    DataFixture
+                    .DefaultIndicatorMeta()
+                    .GenerateList(3)
+                )
+                .WithTimePeriodMetas(() =>
+                    DataFixture
+                    .DefaultTimePeriodMeta()
+                    .GenerateList(3)
+                )
+                .WithMetaSummary();
+
+            await TestApp.AddTestData<PublicDataDbContext>(context => context.DataSetVersions.Add(dataSetVersion));
+
+            var response = await GetMeta(dataSet.Id, dataSetVersion.Version);
+
+            var content = response.AssertOk<DataSetMetaViewModel>(useSystemJson: true);
+
+            Assert.NotNull(content);
+
+            foreach (var filterMetaViewModel in content.Filters)
+            {
+                var filterMeta = Assert.Single(dataSetVersion.FilterMetas, fm => fm.PublicId == filterMetaViewModel.Id);
+                Assert.Equal(filterMeta.Hint, filterMetaViewModel.Hint);
+                Assert.Equal(filterMeta.Label, filterMetaViewModel.Label);
+
+                var allFilterMetaLinks = filterMeta.Options.SelectMany(fom => fom.MetaLinks);
+
+                foreach (var filterOptionMetaViewModel in filterMetaViewModel.Options)
+                {
+                    var filterOptionMetaLink = Assert.Single(
+                        allFilterMetaLinks,
+                        foml => SqidEncoder.Encode(foml.PublicId) == filterOptionMetaViewModel.Id);
```

Comment:
Minor, but this comes up as a typo in Rider because it's 4 letters. Might want to name it `link` instead

Similar thing can be done below as well where we've used `foml` elsewhere (it would should have been `loml` for locations)

Reply:
sorted :)